sdk/cosmos/azure-cosmos-spark_3-0_2-12/src/main/scala/com/azure/cosmos/spark/CosmosPartitionReader.scala

```diff
@@ -25,10 +30,16 @@ case class CosmosPartitionReader(config: Map[String, String], readSchema: Struct
 
   // TODO: moderakh cache the cosmos clients and manage the lifetime of the clients
   // we shouldn't recreate everytime, causing resource leak, inefficient behaviour
-  val cosmosAsyncContainer = new CosmosClientBuilder()
+  val builder = new CosmosClientBuilder()
     .endpoint(endpointConfig.endpoint)
     .key(endpointConfig.key)
-    .buildAsyncClient()
+
+  val state = cosmosClientStateHandle.value;
+
+  CosmosBridgeInternal.metadataCaches(builder, cosmosClientStateHandle.value)
```

Comment:
Just my 2 cents: I don't think the Spark connector should be allowed to use internal API. Instead I think we need to add this to public surface area (as beta first). It encourages customers to use internal API if we start using it across different maven packages/binaries